samples/apps/copilot-chat-app/WebApp/src/libs/auth/AuthHelper.ts

```diff
@@ -75,9 +76,16 @@ const logoutAsync = async (instance: IPublicClientApplication) => {
 // SKaaS = Semantic Kernel as a Service
 // Gets token with scopes to authorize SKaaS specifically
 const getSKaaSAccessToken = async (instance: IPublicClientApplication) => {
-    return instance.acquireTokenSilent({ scopes: Constants.msal.skScopes, account: instance.getAllAccounts()[0] }).then((token) => {
+    try {
+        const token = await instance.acquireTokenSilent({ scopes: Constants.msal.skScopes, account: instance.getAllAccounts()[0] });
         return token.accessToken;
-    });
+    } catch (ex: any) {
+        if (ex instanceof InteractionRequiredAuthError) {
+            const token = await instance.acquireTokenPopup({ scopes: Constants.msal.skScopes, account: instance.getAllAccounts()[0] });
+            return token.accessToken;
+        }
+    }
+    throw new Error('Unable to get SKaaS access token');
```

Comment:
If this message is seen by the user, let's make it a little more friendly: "Failed to get access token for web service."